Report the UGI user in FileStatus instead of process owner.

sidseth opened the pull request:
Hadoop uses UserGroupInformation all over the place to keep track of the current user.

Cloud Storage can't really report any actual owner. However, instead of reporting the unix process owner in FileStatus - there's some small benefits of reporting the UGI user. 
1. There's a bunch of places where permission checks are done. For processes which run on behalf of other users (HiveServer2, Oozie) - these checks can end up failing. A workaround is to set reported permissions to 777.
2. The other HCFS implementations work with UGI - it's good to have consistent behaviour across these.

medb commented on gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java:
```diff
@@ -117,4 +117,9 @@ public void testFileSystemIsRemovedFromCacheOnClose()
       throws IOException {
   }
+
+  @Test @Override
```
Please, replace with:
```
public void testFileStatusUser() throws IOException {}
```

medb commented:
Also, please add a change note to [CHANGES.txt](https://github.com/GoogleCloudPlatform/bigdata-interop/blob/master/gcs/CHANGES.txt)

sidseth commented on gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java:
```diff
@@ -1682,9 +1681,11 @@ public Path getHomeDirectory() {
+  private FileStatus getFileStatus(FileInfo fileInfo) throws IOException {
     // GCS does not provide modification time. It only provides creation time.
     // It works for objects because they are immutable once created.
+    UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
```
I've changed the patch to reduce the number of invocations for list APIs. I'm not sure how big a performance penalty this is. For HDFS - the access(..) method invokes this each time. Getting the current user is also standard practice all over hadoop code.

sidseth commented:
Thanks for taking a look. Updated the PR, and rebased to master. Just got a confirmation that the CLA has been signed. (I signed it! - in case this triggers a recheck)

medb commented:
I see Hortonworks in corporate CLAs signers now.

I guess you need to post standalone "I signed it!" comment to trigger check by bot.

medb commented on gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java:
```diff
@@ -1665,6 +1666,12 @@ String trimToPrefixWithoutGlob(String path) {
   }
 
+  /** Helper method to get the UGI short user name */
+  private String getUgiUserName() throws IOException {
```
Make this method static - it doesn't access any object state.

medb commented:
Merged: https://github.com/GoogleCloudPlatform/bigdata-interop/commit/e538ae0e7db682640f823f12c1226c2972ee9b9e
